CLI: Pass --quiet to disable HMR logging in browser console

niedzielski opened the pull request:
Issue: #8461

Pass the CLI `--quiet` option to the hot module replacement client
instead of hardcoding it in development builds. A minor improvement to
60961d380ed4b132c690e3fef0f7c0ce19d3f6c3 (#9535).

https://github.com/storybookjs/storybook/issues/8461#issuecomment-640515472

## What I did

Pass `--quiet` to webpack-hot-middleware/client. Default to true.

## How to test

Pass `--quiet` to `start-storybook` and HMR messages should disappear.

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this.

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

niedzielski committed:
Pass --quiet to HMR logging in browser console (#8461)

Pass the CLI `--quiet` option to the hot module replacement client
instead of hardcoding it in development builds. A minor improvement to
60961d380ed4b132c690e3fef0f7c0ce19d3f6c3 (#9535).

https://github.com/storybookjs/storybook/issues/8461#issuecomment-640515472

niedzielski commented:
I believe this is a false positive:

```
npm notice created a lockfile as package-lock.json. You should commit this file.


Failing the build, there is 1 fail.
Request failed [403]: https://api.github.com/repos/storybookjs/storybook/issues/11087/comments
Response: {
  "message": "Resource not accessible by integration",
  "documentation_url": "https://developer.github.com/v3/issues/comments/#create-a-comment"
}
Feedback: undefined
```

shilman reviewed (approved):
This looks reasonable to me. I'm assuming errors still show up?

niedzielski commented:
@shilman, I should have done more research. Sorry about that. What I'm seeing is that this will [suppress all warnings too](https://github.com/webpack-contrib/webpack-hot-middleware/blob/930157c5aa0db965166b22c76eb673601140c6db/client.js#L50). I believe errors are always logged but there's some inconsistencies in the logging (e.g., see the [warning checking in process-update](https://github.com/webpack-contrib/webpack-hot-middleware/blob/930157c5aa0db965166b22c76eb673601140c6db/process-update.js#L17)).

There's a [`noInfo` option](https://github.com/webpack-contrib/webpack-hot-middleware#client) I could use instead that _will_ still present warnings. Would that be preferable?

shilman commented:
Try it? As long as it’s opt in, we can do whatever you think is most useful

niedzielski committed:
Allow HMR warnings when --quiet in browser console (#8461)

Revise --quiet to allow hot module replacement warnings to be logged.
webpack-hot-middleware's `quiet` parameter suppress warnings so `noInfo`
is used instead.

niedzielski commented:
Thanks, @shilman! I've passed the quiet option to `noInfo` instead in a new commit as both warnings and errors are useful to me. Feel free to revert if that's ultimately unwanted.

shilman commented:
Looks good @niedzielski! I'll let @ndelangen take a look before merging
